Won't compile using g++-9.

Using g++-8 is fine, switching to g++-9 `g++ (Debian 9.2.1-1) 9.2.1 20190813` results in:

```
In file included from ./submodules/spirit-po/include/spirit_po/default_plural_forms_compiler.hpp:68,
                 from ./submodules/spirit-po/include/spirit_po/catalog.hpp:23,
                 from ./submodules/spirit-po/include/spirit_po/spirit_po.hpp:10,
                 from src/localization.hpp:6,
                 from src/localization_bindings.cpp:1:
./submodules/spirit-po/include/spirit_po/default_plural_forms_expressions.hpp: In member function ‘std::vector<boost::variant<spirit_po::default_plural_forms::constant, spirit_po::default_plural_forms::skip, spirit_po::default_plural_forms::skip_if, spirit_po::default_plural_forms::skip_if_not, spirit_po::default_plural_forms::op_code> > spirit_po::default_plural_forms::emitter::operator()(const spirit_po::default_plural_forms::and_op&) const’:
./submodules/spirit-po/include/spirit_po/default_plural_forms_expressions.hpp:386:62: error: ‘constexpr boost::static_visitor<R>::static_visitor() [with R = bool]’ is protected within this context
  386 |     bool second_is_boolean = boost::apply_visitor(is_boolean{}, o.e2);
      |                                                              ^
In file included from /usr/include/boost/variant/detail/hash_variant.hpp:22,
                 from /usr/include/boost/variant/variant.hpp:34,
                 from /usr/include/boost/spirit/home/support/info.hpp:14,
                 from /usr/include/boost/spirit/home/qi/domain.hpp:16,
                 from /usr/include/boost/spirit/home/qi/meta_compiler.hpp:15,
                 from /usr/include/boost/spirit/home/qi/action/action.hpp:14,
                 from /usr/include/boost/spirit/home/qi/action.hpp:14,
                 from /usr/include/boost/spirit/home/qi.hpp:14,
                 from /usr/include/boost/spirit/include/qi.hpp:16,
                 from ./submodules/spirit-po/include/spirit_po/catalog_metadata.hpp:15,
                 from ./submodules/spirit-po/include/spirit_po/catalog.hpp:22,
                 from ./submodules/spirit-po/include/spirit_po/spirit_po.hpp:10,
                 from src/localization.hpp:6,
                 from src/localization_bindings.cpp:1:
/usr/include/boost/variant/static_visitor.hpp:52:5: note: declared protected here
   52 |     static_visitor() = default;
      |     ^~~~~~~~~~~~~~
In file included from ./submodules/spirit-po/include/spirit_po/default_plural_forms_compiler.hpp:68,
                 from ./submodules/spirit-po/include/spirit_po/catalog.hpp:23,
                 from ./submodules/spirit-po/include/spirit_po/spirit_po.hpp:10,
                 from src/localization.hpp:6,
                 from src/localization_bindings.cpp:1:
./submodules/spirit-po/include/spirit_po/default_plural_forms_expressions.hpp:386:62: error: ‘boost::static_visitor<R>::~static_visitor() [with R = bool]’ is protected within this context
  386 |     bool second_is_boolean = boost::apply_visitor(is_boolean{}, o.e2);
      |                                                              ^
In file included from /usr/include/boost/variant/detail/hash_variant.hpp:22,
                 from /usr/include/boost/variant/variant.hpp:34,
                 from /usr/include/boost/spirit/home/support/info.hpp:14,
                 from /usr/include/boost/spirit/home/qi/domain.hpp:16,
                 from /usr/include/boost/spirit/home/qi/meta_compiler.hpp:15,
                 from /usr/include/boost/spirit/home/qi/action/action.hpp:14,
                 from /usr/include/boost/spirit/home/qi/action.hpp:14,
                 from /usr/include/boost/spirit/home/qi.hpp:14,
                 from /usr/include/boost/spirit/include/qi.hpp:16,
                 from ./submodules/spirit-po/include/spirit_po/catalog_metadata.hpp:15,
                 from ./submodules/spirit-po/include/spirit_po/catalog.hpp:22,
                 from ./submodules/spirit-po/include/spirit_po/spirit_po.hpp:10,
                 from src/localization.hpp:6,
                 from src/localization_bindings.cpp:1:
/usr/include/boost/variant/static_visitor.hpp:53:5: note: declared protected here
   53 |     ~static_visitor() = default;
      |     ^
In file included from ./submodules/spirit-po/include/spirit_po/default_plural_forms_compiler.hpp:68,
                 from ./submodules/spirit-po/include/spirit_po/catalog.hpp:23,
                 from ./submodules/spirit-po/include/spirit_po/spirit_po.hpp:10,
                 from src/localization.hpp:6,
                 from src/localization_bindings.cpp:1:
./submodules/spirit-po/include/spirit_po/default_plural_forms_expressions.hpp: In member function ‘std::vector<boost::variant<spirit_po::default_plural_forms::constant, spirit_po::default_plural_forms::skip, spirit_po::default_plural_forms::skip_if, spirit_po::default_plural_forms::skip_if_not, spirit_po::default_plural_forms::op_code> > spirit_po::default_plural_forms::emitter::operator()(const spirit_po::default_plural_forms::or_op&) const’:
./submodules/spirit-po/include/spirit_po/default_plural_forms_expressions.hpp:407:62: error: ‘constexpr boost::static_visitor<R>::static_visitor() [with R = bool]’ is protected within this context
  407 |     bool second_is_boolean = boost::apply_visitor(is_boolean{}, o.e2);
      |                                                              ^
In file included from /usr/include/boost/variant/detail/hash_variant.hpp:22,
                 from /usr/include/boost/variant/variant.hpp:34,
                 from /usr/include/boost/spirit/home/support/info.hpp:14,
                 from /usr/include/boost/spirit/home/qi/domain.hpp:16,
                 from /usr/include/boost/spirit/home/qi/meta_compiler.hpp:15,
                 from /usr/include/boost/spirit/home/qi/action/action.hpp:14,
                 from /usr/include/boost/spirit/home/qi/action.hpp:14,
                 from /usr/include/boost/spirit/home/qi.hpp:14,
                 from /usr/include/boost/spirit/include/qi.hpp:16,
                 from ./submodules/spirit-po/include/spirit_po/catalog_metadata.hpp:15,
                 from ./submodules/spirit-po/include/spirit_po/catalog.hpp:22,
                 from ./submodules/spirit-po/include/spirit_po/spirit_po.hpp:10,
                 from src/localization.hpp:6,
                 from src/localization_bindings.cpp:1:
/usr/include/boost/variant/static_visitor.hpp:52:5: note: declared protected here
   52 |     static_visitor() = default;
      |     ^~~~~~~~~~~~~~
In file included from ./submodules/spirit-po/include/spirit_po/default_plural_forms_compiler.hpp:68,
                 from ./submodules/spirit-po/include/spirit_po/catalog.hpp:23,
                 from ./submodules/spirit-po/include/spirit_po/spirit_po.hpp:10,
                 from src/localization.hpp:6,
                 from src/localization_bindings.cpp:1:
./submodules/spirit-po/include/spirit_po/default_plural_forms_expressions.hpp:407:62: error: ‘boost::static_visitor<R>::~static_visitor() [with R = bool]’ is protected within this context
  407 |     bool second_is_boolean = boost::apply_visitor(is_boolean{}, o.e2);
      |                                                              ^
In file included from /usr/include/boost/variant/detail/hash_variant.hpp:22,
                 from /usr/include/boost/variant/variant.hpp:34,
                 from /usr/include/boost/spirit/home/support/info.hpp:14,
                 from /usr/include/boost/spirit/home/qi/domain.hpp:16,
                 from /usr/include/boost/spirit/home/qi/meta_compiler.hpp:15,
                 from /usr/include/boost/spirit/home/qi/action/action.hpp:14,
                 from /usr/include/boost/spirit/home/qi/action.hpp:14,
                 from /usr/include/boost/spirit/home/qi.hpp:14,
                 from /usr/include/boost/spirit/include/qi.hpp:16,
                 from ./submodules/spirit-po/include/spirit_po/catalog_metadata.hpp:15,
                 from ./submodules/spirit-po/include/spirit_po/catalog.hpp:22,
                 from ./submodules/spirit-po/include/spirit_po/spirit_po.hpp:10,
                 from src/localization.hpp:6,
                 from src/localization_bindings.cpp:1:
/usr/include/boost/variant/static_visitor.hpp:53:5: note: declared protected here
   53 |     ~static_visitor() = default;
      |     ^
In file included from ./submodules/spirit-po/include/spirit_po/default_plural_forms_compiler.hpp:68,
                 from ./submodules/spirit-po/include/spirit_po/catalog.hpp:23,
                 from ./submodules/spirit-po/include/spirit_po/spirit_po.hpp:10,
                 from src/localization.hpp:6,
                 from src/localization_bindings.cpp:1:
./submodules/spirit-po/include/spirit_po/default_plural_forms_expressions.hpp: In constructor ‘spirit_po::default_plural_forms::stack_machine::stack_machine(const expr&)’:
./submodules/spirit-po/include/spirit_po/default_plural_forms_expressions.hpp:494:53: error: ‘constexpr boost::static_visitor<R>::static_visitor() [with R = std::vector<boost::variant<spirit_po::default_plural_forms::constant, spirit_po::default_plural_forms::skip, spirit_po::default_plural_forms::skip_if, spirit_po::default_plural_forms::skip_if_not, spirit_po::default_plural_forms::op_code> >]’ is protected within this context
  494 |     : instruction_seq_(boost::apply_visitor(emitter{}, e))
      |                                                     ^
In file included from /usr/include/boost/variant/detail/hash_variant.hpp:22,
                 from /usr/include/boost/variant/variant.hpp:34,
                 from /usr/include/boost/spirit/home/support/info.hpp:14,
                 from /usr/include/boost/spirit/home/qi/domain.hpp:16,
                 from /usr/include/boost/spirit/home/qi/meta_compiler.hpp:15,
                 from /usr/include/boost/spirit/home/qi/action/action.hpp:14,
                 from /usr/include/boost/spirit/home/qi/action.hpp:14,
                 from /usr/include/boost/spirit/home/qi.hpp:14,
                 from /usr/include/boost/spirit/include/qi.hpp:16,
                 from ./submodules/spirit-po/include/spirit_po/catalog_metadata.hpp:15,
                 from ./submodules/spirit-po/include/spirit_po/catalog.hpp:22,
                 from ./submodules/spirit-po/include/spirit_po/spirit_po.hpp:10,
                 from src/localization.hpp:6,
                 from src/localization_bindings.cpp:1:
/usr/include/boost/variant/static_visitor.hpp:52:5: note: declared protected here
   52 |     static_visitor() = default;
      |     ^~~~~~~~~~~~~~
In file included from ./submodules/spirit-po/include/spirit_po/default_plural_forms_compiler.hpp:68,
                 from ./submodules/spirit-po/include/spirit_po/catalog.hpp:23,
                 from ./submodules/spirit-po/include/spirit_po/spirit_po.hpp:10,
                 from src/localization.hpp:6,
                 from src/localization_bindings.cpp:1:
./submodules/spirit-po/include/spirit_po/default_plural_forms_expressions.hpp:494:53: error: ‘boost::static_visitor<R>::~static_visitor() [with R = std::vector<boost::variant<spirit_po::default_plural_forms::constant, spirit_po::default_plural_forms::skip, spirit_po::default_plural_forms::skip_if, spirit_po::default_plural_forms::skip_if_not, spirit_po::default_plural_forms::op_code> >]’ is protected within this context
  494 |     : instruction_seq_(boost::apply_visitor(emitter{}, e))
      |                                                     ^
In file included from /usr/include/boost/variant/detail/hash_variant.hpp:22,
                 from /usr/include/boost/variant/variant.hpp:34,
                 from /usr/include/boost/spirit/home/support/info.hpp:14,
                 from /usr/include/boost/spirit/home/qi/domain.hpp:16,
                 from /usr/include/boost/spirit/home/qi/meta_compiler.hpp:15,
                 from /usr/include/boost/spirit/home/qi/action/action.hpp:14,
                 from /usr/include/boost/spirit/home/qi/action.hpp:14,
                 from /usr/include/boost/spirit/home/qi.hpp:14,
                 from /usr/include/boost/spirit/include/qi.hpp:16,
                 from ./submodules/spirit-po/include/spirit_po/catalog_metadata.hpp:15,
                 from ./submodules/spirit-po/include/spirit_po/catalog.hpp:22,
                 from ./submodules/spirit-po/include/spirit_po/spirit_po.hpp:10,
                 from src/localization.hpp:6,
                 from src/localization_bindings.cpp:1:
/usr/include/boost/variant/static_visitor.hpp:53:5: note: declared protected here
   53 |     ~static_visitor() = default;
      |     
```

Boost Version: 1.67.0.2

--- a/include/spirit_po/default_plural_forms_expressions.hpp
+++ b/include/spirit_po/default_plural_forms_expressions.hpp
@@ -383,7 +383,7 @@ FOREACH_SPIRIT_PO_BINARY_OP(EMIT_OP_)
   std::vector<instruction> operator()(const and_op & o) const {
     auto result = boost::apply_visitor(*this, o.e1);
     auto second = boost::apply_visitor(*this, o.e2);
-    bool second_is_boolean = boost::apply_visitor(is_boolean{}, o.e2);
+    bool second_is_boolean = boost::apply_visitor(is_boolean(), o.e2);
 
     uint sec_size = static_cast<uint>(second.size());
     if (!second_is_boolean) { sec_size += 2; }
@@ -404,7 +404,7 @@ FOREACH_SPIRIT_PO_BINARY_OP(EMIT_OP_)
   std::vector<instruction> operator()(const or_op & o) const {
     auto result = boost::apply_visitor(*this, o.e1);
     auto second = boost::apply_visitor(*this, o.e2);
-    bool second_is_boolean = boost::apply_visitor(is_boolean{}, o.e2);
+    bool second_is_boolean = boost::apply_visitor(is_boolean(), o.e2);
 
     uint sec_size = static_cast<uint>(second.size());
     if (!second_is_boolean) { sec_size += 2; }
@@ -491,7 +491,7 @@ private:
 
 public:
   explicit stack_machine(const expr & e)
-    : instruction_seq_(boost::apply_visitor(emitter{}, e))
+    : instruction_seq_(boost::apply_visitor(emitter(), e))
     , stack_()
     , n_value_()
   {}

